core/src/main/java/de/danoeh/antennapod/core/feed/Feed.java

```diff
@@ -439,12 +465,55 @@ public void setFeedIdentifier(String feedIdentifier) {
         this.feedIdentifier = feedIdentifier;
     }
 
-    public String getPaymentLink() {
-        return paymentLink;
+    public ArrayList<Funding> extractPaymentLinks(String payLinks) {
+        if (StringUtil.isBlank(payLinks)) {
+            return null;
+        }
+        // old format before we started storing the urls as pay= and fund=
+        fundingList = new ArrayList<Funding>();
+        if (!payLinks.contains(SUPPORT_INTERNAL_SPLIT) && !payLinks.contains(SUPPORT_INTERNAL_EQUAL)) {
+            fundingList.add(new Funding(payLinks, "Support this Podcast"));
+            return fundingList;
+        }
+        String [] list = payLinks.split(SUPPORT_INTERNAL_SPLIT);
+        if (list.length == 0) {
+            return null;
+        }
+
+        for (String str : list) {
+            try {
+                String [] linkContent = str.split(SUPPORT_INTERNAL_EQUAL);
+                fundingList.add(new Funding(linkContent[0], linkContent[1]));
+            } catch (ArrayIndexOutOfBoundsException e) {
+                Log.d(TAG, "Payment link in the wrong format" + payLinks);
```

Comment:
Does this actually happen when we don't do something really wrong when storing the items? Not sure if the precaution here is needed.